lookout/style/format/postprocess.py

```diff
@@ -104,35 +113,51 @@ def filter_uast_breaking_preds(
                            left and right babelfish nodes.
     :param node_parents: Parents mapping of the input UASTs.
     :param rule_winners: Numpy array of the index of the winning rule for each sample.
-    :param log: Logger.
+    :param grouped_quote_predictions: Quotes predictions (handled differenlty from the rest).
     :return: List of predictions indices that are considered valid i.e. that are not breaking
              the UAST.
     """
-    CLS_QUOTES = {CLS_SINGLE_QUOTE, CLS_DOUBLE_QUOTE}
     safe_preds = []
     current_path = None  # type: Optional[str]
     parsing_cache = {}  # type: Dict[int, Optional[Tuple[bblfsh.Node, int, int]]]
+    file_content = None  # type: Optional[str]
     cur_i = 0
     for i, (gt, pred, vn_y) in enumerate(zip(y, y_pred, vnodes_y)):
         if vn_y.path != current_path:
             parsing_cache = {}
             current_path = vn_y.path
+            file_content = files[vn_y.path].content.decode("utf-8", "replace")
         while vn_y is not vnodes[cur_i]:
             cur_i += 1
             if cur_i >= len(vnodes):
                 raise AssertionError("vnodes_y and vnodes are not consistent.")
-        if gt == pred:
-            safe_preds.append(i)
+        # quote types are special cased
+        if id(vn_y) in grouped_quote_predictions:
+            pred_string = _prediction_to_text(pred, feature_extractor)
+            group = grouped_quote_predictions[id(vn_y)]
+            # already handled with the previous vnode
+            if group is None:
+                continue
+            vnode1, vnode2, vnode3 = group
+            content1, content2, content3 = map(lambda v: file_content[v.start.offset:v.end.offset],
+                                               (vnode1, vnode2, vnode3))
```

Comment:
Why not `content1, content2, content3 =  vnode1.value, vnode2.value, vnode3.value`? should be the same.